pkg/controller/customdomain/customdomain_utils.go

```diff
@@ -98,12 +100,39 @@ func FindCustomDomainCondition(conditions []customdomainv1alpha1.CustomDomainCon
 	return nil
 }
 
-// finalizeCustomDomain cleans up once a CustomDomain CR is deleted
+// finalizeCustomDomain cleans up left over resources a CustomDomain CR is deleted
 func (r *ReconcileCustomDomain) finalizeCustomDomain(reqLogger logr.Logger, instance *customdomainv1alpha1.CustomDomain) error {
-	// restore the custom domain
-	err := modifyClusterDomain(r, reqLogger, instance, true)
+	// get and delete the secret in openshift-ingress
+	ingressSecret := &corev1.Secret{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{
+		Namespace: ingressNamespace,
+		Name:      instance.Name,
+	}, ingressSecret)
 	if err != nil {
-		return err
+		reqLogger.Error(err, fmt.Sprintf("Failed to get %s secret", instance.Name))
+	} else {
+		err = r.client.Delete(context.TODO(), ingressSecret)
```

Comment:
It looks like we're [not necessarily creating](https://github.com/openshift/custom-domains-operator/pull/11/files#diff-71f68d48c4d024a88d5d71f8b173da8e4319b5e86ca6901c77e70c4b4240b4a4R185) this secret. Do we really want to delete it if it existed before we got here?

I guess since we're effectively replacing it, we effectively own it?

Reply:
Yes, am I missing something? Is there a way to "own" resources?

Reply:
Not across namespaces, unfortunately.

Reading the code, it's not obvious whether we expect the ingress secret and/or IngressController CR to always have been created by this operator. If that's always true, then this is okay -- though I would still prefer to be more aggressive about the deletion (fail/retry if `Delete` fails, kind of thing) and then less forgiving about the creation (nix the `Update` path, just use the `Create` and let it fail if the object exists).

Reply:
Took your advice and nix'd the Update path. I'll need to think about being more aggressive on the delete.